processor/src/main/java/org/mapstruct/ap/internal/model/source/Mapping.java

```diff
@@ -343,11 +345,13 @@ public void init(SourceMethod method, FormattingMessager messager, TypeFactory t
      * @param method the source method that the mapping belongs to
      * @param messager the messager that can be used for outputting messages
      * @param typeFactory the type factory
+     * @param namingUtils the accessor naming utils
      * @param isReverse whether the init is for a reverse mapping
      * @param reverseSourceParameter the source parameter from the revers mapping
      */
-    private void init(SourceMethod method, FormattingMessager messager, TypeFactory typeFactory, boolean isReverse,
-        Parameter reverseSourceParameter) {
+    private void init(SourceMethod method, FormattingMessager messager, TypeFactory typeFactory,
+                      AccessorNamingUtils namingUtils, boolean isReverse,
```

Comment:
Now this is an instance.. should we somehow combine this with the `CollectionMappingStrategy`? Feels like this is very much related and in the past I kept that one dragging around.. Anyway.. perhaps solve that separately.

Reply:
What do you mean combine it with the `CollectionMappingStrategy`? I don't get it 😄

Reply:
Well, for collections which method to select in the end comes down to both: `CollectionMappingStrategy` and `AccessorNamingUtils`. Now they have to be combined on the call site.

Reply:
They were always combined. Sorry if I don't quite follow